### DISCUSSION TITLE
Accessing all possible short forecasts

### DISCUSSION THREAD

**Discussion by nateHopper** [weather-gov/api#453] [Q&A]
Hello!

I'm currently developing a weather webapp and I wanted to do something similar to the Timeline underneath the current forecast in [Dark Sky](https://darksky.net)

For this, I wanted to make a switch statement to assign a different color and text to the bar segments depending on the short forecast. With this, though, I need to be able to access all of the possible short forecasts in the API. Is there any way for me to access a list or directory with all of the possible JSON responses for "shortForecast"?

Thank you in advance.

**Comment by scadergit** [comment]
This is a popular request (used be pinned when we used issues), but the short answer is that it is not advised to key on the short forecast. The short forecast (similar to the description) is generated using NWS directives that can change to improve data quality. We plan to add a keyable property in a future release, which would work similar to how the icon urls work today.

**Reply by jasonpaulmichel** [reply]
Any insight into where the keyable property feature is on your roadmap?

**Reply by StephenClouse** [reply]
We do not currently have any timetable for this change.

**Reply by scadergit** [reply]
fwiw, I reminded mgmt of this need, even for our own use of the API.

**Reply by jasonpaulmichel** [reply]
Thank you for the reply!

**Reply by austinginn** [reply]
I was about to ask this same question.  Any update on the timeline? Thanks!

**Comment by StephenClouse** [comment]
More to the point from what @scadergit said, the short forecast text is dynamically composed and can contain millions of different variations based on coverage, intensity, and timing. It's not a simple static list we're pulling from.

**Comment by netbymatt** [comment]
I was reviewing the data provided at the gridpoints endpoint (i.e. https://api.weather.gov/gridpoints/MLB/26,68) and noticed some data I hadn't seen before in `properties.weather[]` which includes time periods in the several hour range with these properties: `coverage`, `weather`, `intensity`, `visibility`, and `attributes`. The live data has one or two word values for each property.

When I dig into the open api specification (https://api.weather.gov/openapi.json) all of these properties have enums listed which tells me that we could use this as a keyable forecast. It's not exactly what we're looking for with `shortForecast` but scanning through this array and picking out the times that are interesting is a fairly straight-forward task. For reference here is an excerpt of the `properties.weather[]` api definition.

```
{
	"coverage":{
		"enum":[
			"areas",
			"brief",
			"chance",
			"definite",
			"few",
			"frequent",
			"intermittent",
			"isolated",
			"likely",
			"numerous",
			"occasional",
			"patchy",
			"periods",
			"scattered",
			"slight_chance",
			"widespread"
		],
		"type":"string",
		"nullable":true
	},
	"weather":{
		"enum":[
			"blowing_dust",
			"blowing_sand",
			"blowing_snow",
			"drizzle",
			"fog",
			"freezing_fog",
			"freezing_drizzle",
			"freezing_rain",
			"freezing_spray",
			"frost",
			"hail",
			"haze",
			"ice_crystals",
			"ice_fog",
			"rain",
			"rain_showers",
			"sleet",
			"smoke",
			"snow",
			"snow_showers",
			"thunderstorms",
			"volcanic_ash",
			"water_spouts"
		],
		"type":"string",
		"nullable":true
	},
	"intensity":{
		"enum":[
			"very_light",
			"light",
			"moderate",
			"heavy"
		],
		"type":"string",
		"nullable":true
	},
	"visibility":{
		"$ref":"#\/components\/schemas\/QuantitativeValue"
	},
	"attributes":{
		"type":"array",
		"items":{
			"enum":[
				"damaging_wind",
				"dry_thunderstorms",
				"flooding",
				"gusty_wind",
				"heavy_rain",
				"large_hail",
				"small_hail",
				"tornadoes"
			],
			"type":"string"
		}
	}
}
```

**Reply by t2** [reply]
@scadergit @StephenClouse Any update on having a keyable enum for weather? We are swapping from OpenWeather and their `icon` property allowed us to get one of these 9 options Sunny, Clear Skies, Few Clouds, Cloudy, Rainy, Stormy, Snowy, Misty, Clear. I would need something similar.
